### ISSUE TITLE
Uncaught TypeError: Cannot read property 'call' of undefined

### ISSUE THREAD

**Issue by kirikintha** [mozilla-neutrino/neutrino-dev#869]
### Bug or issue?

Issue with resolution

Please try to answer the following questions:

- What version of Neutrino are you using?

`    "neutrino": "^8.2.3",`

- Are you trying to use any presets? If so, which ones, and what versions?

```
    "@neutrinojs/airbnb": "^8.2.3",
    "@neutrinojs/copy": "^8.2.3",
    "@neutrinojs/dev-server": "^8.2.3",
    "@neutrinojs/eslint": "^8.2.3",
    "@neutrinojs/html-template": "^8.2.3",
    "@neutrinojs/react": "^8.2.3",
    "@neutrinojs/style-loader": "^8.2.3",

```
- Are you using the Yarn client or the npm client? What version?

NPM

- What version of Node.js are you using?

8.1.4

- What operating system are you using?

OSX High Sierra

- What did you do?

Ran neutrino build with the following:

```

// React
    ['@neutrinojs/react', {
      targets: {
        browsers: [
          'last 2 versions',
          'not ie < 11',
        ],
      },
    }],
    // HTML Template
    ['@neutrinojs/html-template', {
      baseHref: '/',
      template: './src/index.html5.ejs',
      appMountId: 'root',
      title: 'eSign Digital Petitions',
      favicon: '',
      xhtml: true,
      mobile: false,
      minify: {
        useShortDoctype: true,
        keepClosingSlash: true,
        collapseWhitespace: true,
        preserveLineBreaks: true,
      },
    }],
    // Style Loader
    ['@neutrinojs/style-loader', {
      loaders: [{
        loader: 'sass-loader',
        useId: 'sass',
      }],
      test: /\.scss$/,
    }],
```

- What did you expect to happen?

I expected the compiled output to work as in previous version.

- What actually happened, contrary to your expectations?

This is a little hard to explain, but I had two components that were failing with:

`Uncaught TypeError: Cannot read property 'call' of undefined`
 
### Feature request or enhancement?

Resolution comments:

This is one of those errors you hate to get in webpack. I have seen this in some really odd cases when using the ES6 import feature with babel, minification and uglification. The crazy thing is: other components with the same boilerplate worked. All internal tests passed, dev server worked, compiled and linted - but these two errant components just refused to load in their chunk in production mode. 

The awful thing about errors like these, is that they are compile issues with treeshaking, chunks and uglifying code in *larger* projects.

I tried a lot of different steps, including making chunks larger, vendoring chunks, moving things around in code, playing with chunk mode - but nothing was solving the issue.

So, I looked at how minification was happening in babel and made this change in my neutrinorc

```
    ['@neutrinojs/minify', {
      babel: {
        mangle: {
          keepFnName: true,
        },
        deadcode: {
          keepFnName: true,
        },
      },
    }],

```

Which immediately solved the issue. So, something in the complexity of the project and mangling the information was mangling the chunked chunks. 

In other words: How many mangles can a mangled code chunk, if a mangled chunk could chunk chunks? 

In case anyone else is having this issue, changing these values upped my index file by about 30k, but in the end solved the issue of crashing the app.

Any other thoughts on this? I am not seeing any side effects, but wanted to gain some more information from the community.

**Comment by edmorley** [comment]
Hi! Sorry for the delayed reply.

This looks like one of those bugs that was a massive pain to debug - sorry for the hassle! Thankfully in Neutrino 9 (currently unreleased, is just on `master`) the whole `@neutrinojs/chunk` has been removed in favour of webpack's new `splitChunks` feature. 

As such, this should be fixed there - though I'm open to ideas as to what we should do for Neutrino 8... :-)

**Comment by eliperelman** [comment]
The problem with going towards a time-based hash is that it will never be persisted, i.e. it was always be cache-busted. It would need to always get a deterministic hash when the content is the same, and only change when it's different in order to rely on the cache.

**Comment by edmorley** [comment]
It's worth noting that the Neutrino 8 chunking middleware uses the Plugins/config recommended here:
https://medium.com/webpack/predictable-long-term-caching-with-webpack-d3eee1d3fa31

...which is linked from the webpack caching docs:
https://webpack.js.org/guides/caching/#conclusion

As such, if it's buggy it would be good to get it fixed upstream.

**Comment by edmorley** [comment]
Closing since this issue no longer exists on `master` (due to the switch to webpack 4's `splitChunks`) and since the Neutrino 8 long term caching behaviour is copied from the webpack docs/blog post linked above -- so any bug really needs to be fixed there.
